os/fs/smartfs/smartfs_smart.c

```diff
@@ -693,6 +715,18 @@ static int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_of
 	}
 #endif							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+	/* File's data sector has been synced with MTD, now check if this is a new file entry and write the entry */
+	if (sf->bflags & SMARTFS_BFLAG_NEW_ENTRY) {
+		ret = smartfs_writeentry(fs, sf->entry, sf->entry.name, SMARTFS_DIRENT_TYPE_FILE, sf->mode, &sf->entry, sf->last_psector);
+		if (ret < 0) {
+			fdbg("Failed to write new file entry ot MTD\n");
+			goto errout;
+		}
+		sf->bflags = 0;
```

Comment:
I think you can make nice name for default one.
#define SMARTFS_BFLAG_XXX  0

Reply:
Ohkay, let's do that.